### ISSUE TITLE
7z stuck on trying the first password while trying 6 digits Iterations then gpu stops

### ISSUE THREAD

**Issue by 0x0OZ** [hashcat/hashcat#4020]
**Describe the bug**
I am trying to crack a 7z hash, hashcat is running but it is taking so long against the first password to try while trying very large iterations

If I keep it and wait, the iteration will reach a large number the GPU will stop working there and only the CPU will start cracking hashes (see progress for GPU in image1 and image2)

![image1](https://github.com/hashcat/hashcat/assets/52073989/8674353c-72ca-48bd-8014-fc611f5b28df)

After it finds some iteration, the GPU usage becomes near 0% and as you can see the GPU platform didn't try any password

![image2](https://github.com/hashcat/hashcat/assets/52073989/68075767-3b4e-4c1d-b8a7-1ede07668808)

**To Reproduce**
[hash_latest_7z.txt](https://github.com/user-attachments/files/15785470/hash_latest_7z.txt)

**Expected behavior**
Hashcat should start trying different passwords instead of being stuck at iterating through the first password

**Hardware/Compute device (please complete the following information):**
```
OpenCL API (OpenCL 3.0 CUDA 12.4.131) - Platform #1 [NVIDIA Corporation]
========================================================================
* Device #1: NVIDIA GeForce RTX 3060 Laptop GPU, 6016/6143 MB (1535 MB allocatable), 30MCU

OpenCL API (OpenCL 3.0 ) - Platform #2 [Intel(R) Corporation]
=============================================================
* Device #2: Intel(R) UHD Graphics, 6464/13012 MB (2047 MB allocatable), 32MCU
```
- OpenCL/CUDA driver source: .exe installer

**Hashcat version (please complete the following information):**
 - OS: Windows 10
 - Version: 10.0.19045 Build 19045

**Diagnostic output compute devices:**

```
For NV: Post nvidia-smi output. This tool also exist on Windows
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 552.22                 Driver Version: 552.22         CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                     TCC/WDDM  | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3060 ...  WDDM  |   00000000:01:00.0 Off |                  N/A |
| N/A   59C    P0             29W /   68W |    5223MiB /   6144MiB |      3%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

```
Post clinfo output

[clinfo.txt](https://github.com/user-attachments/files/15785785/clinfo.txt)

Post hashcat -I output

[hashcat-i.txt](https://github.com/user-attachments/files/15785792/hashcat-i.txt)

**Additional context**
The 7z file is around 9 years old, I used 7zhashcat and 7zjohn and got the same resulting hash.

**Comment by PenguinKeeper7** [comment]
> GPU will stop working there and only the CPU will start cracking hashes 

Yes, this is intentional. 7z is one of few modules where part of the work is done on the GPU, which is usually fast then the second stage is on the CPU which is a lot slower. It happens with every 7-Zip hash. 

> Hashcat should start trying different passwords instead of being stuck at iterating through the first password

It is using different passwords, you can see the "Progress" number go up in each screenshot and the Candidates are changing

**Comment by 0x0OZ** [comment]
> > GPU will stop working there and only the CPU will start cracking hashes
> 
> Yes, this is intentional. 7z is one of few modules where part of the work is done on the GPU, which is usually fast then the second stage is on the CPU which is a lot slower. It happens with every 7-Zip hash.
> 
> > Hashcat should start trying different passwords instead of being stuck at iterating through the first password
> 
> It is using different passwords, you can see the "Progress" number go up in each screenshot and the Candidates are changing

The 2nd Candidate (CPU ) is changing, while the first (GPU) is not that's what I am trying to report about

**Comment by matrix** [comment]
please use forum or discord for this
